pkg/kubectl/cmd/config/use_context.go

```diff
@@ -82,10 +84,16 @@ func (o *useContextOptions) complete(cmd *cobra.Command) bool {
 	return true
 }
 
-func (o useContextOptions) validate() error {
+func (o useContextOptions) validate(config *clientcmdapi.Config) error {
 	if len(o.contextName) == 0 {
 		return errors.New("You must specify a current-context")
 	}
 
-	return nil
+	for _, context := range config.Contexts {
+		if context.Cluster == o.contextName {
```

Comment:
Ok that's confusing. I thought we were using NamedContext but it doesn't look like that's in the v1 clientcmd api https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/client/clientcmd/api/v1/types.go#L40.

If config.Contexts is a map, then yes, the key is the context name.

@deads2k how does kubectl determine which clientcmd api version it's using?